js/src/gtag-events/index.js

```diff
@@ -66,7 +66,7 @@ const singleAddToCartClick = function ( event ) {
 };
 
 // Register for add_to_cart click events.
-window.onload = function () {
+window.addEventListener( 'load', function () {
```

Comment:
To make linter happy, and to execute once content is there (as suggested by @eason9487 
```suggestion
document.defaultView.addEventListener( 'DOMContentLoaded', function () {
```

Reply:
Would you suggest to make the same change here: https://github.com/woocommerce/google-listings-and-ads/blob/fix/1630-onload-conflict/src/Google/GlobalSiteTag.php#L465

Reply:
I think so.

However, I'm still not sure about the timing of all those `maybe_display_conversion_and_purchase_event_snippets`. Are we guaranteed they are stamped before the document loads? (So we won't attach the event listener after it was fired)

Reply:
Those events update some order meta when the page is generated in PHP so it isn't tracked a second time. We will be sending the event a little later then before, but previously it was also a problem that even though we mark it as tracked we can't guarantee that the event actually got sent when the page was loaded.
It's the same pattern that's used in Google Analytics Integration, so it is widely used.

Reply:
> So we won't attach the event listener after it was fired

The function `print_inline_gtag_script` wraps the event code in the onload function, so it can't fire the event before the listener is attached.

Reply:
I meant a different thing.

My question was about the timing of `maybe_display_conversion_and_purchase_event_snippets` itself. When it is called, are we sure it's not called after the content is loaded. In cases like the streamed HTML, or some other tricks to insert an executable `script` tag.
Then the timing would be:
1. Script A does `window.addEventListener('DOMContentLoaded', doSmth );`
2. Content is loaded
3. `doSmth` is executed
4. Script B does `window.addEventListener('DOMContentLoaded', doSmthElse );`

Then `doSmthElse` is never fired

Reply:
But I guess we don't do much magic about those scripts and just concatenate them to the returned document, right?

Reply:
Yes it's printed directly in the document. Although there are plugins that can compress / lazy load inline scripts as well do you think in that case it's safer to not wrap the inline scripts?